fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/ByteScanLimiter.java

```diff
@@ -76,8 +91,54 @@ public long getLimit() {
         return originalLimit;
     }
 
+    /**
+     * Returns the number of bytes that have been scanned thus far.
+     *
+     * @return the number of bytes that have been scanned
+     */
+    public long getBytesScanned() {
+        return originalLimit - bytesRemaining.get();
+    }
+
     @Override
     public String toString() {
         return String.format("ByteScanLimiter(%d limit, %d left)", originalLimit, bytesRemaining.get());
     }
+
+    /**
+     * A non-tracking, non-enforcing limiter.
+     */
+    protected static class Untracked extends ByteScanLimiter {
```

Comment:
Is there a reason (other than backwards compatibility) to have `Untracked` extend `ByteScanLimiter`, rather than making `ByteScanLimiter` an interface with two implementations? I guess it saves us a virtual dispatch?

Reply:
Because I was curious, I did some digging: https://stackoverflow.com/questions/23948009/whats-the-size-cost-of-java-inheritance.  If the interweebs are to be believed, then there is little to no cost overhead for method dispatch through an inheritance tree.